crates/sui/src/upgrade_compatibility.rs

```diff
@@ -1301,6 +1304,58 @@ fn struct_ability_mismatch_diag(
     Ok(diags)
 }
 
+fn field_to_string(field: &Field) -> (String, String, String) {
```

Comment:
Consider adding a `FieldInfo` struct to make it a bit more clean to work with these fields' values.